### PR TITLE
ISSUE #155: useQuorum parameters added to check if quorum is enabled/…

### PR DESCRIPTION
**Pull request by tw185112**
…disabled to stop volume attachments from failing

Signed-off-by: Tom Wilson <thomas.wilson@ncr.com>

### CODE REVIEW

File: pkg/client/linstor.go [hunk 1]
```diff
@@ -488,10 +495,24 @@ func (s *Linstor) Attach(ctx context.Context, volId, node string, readOnly bool)
...
+			return nil
+		} 
+		for k, v := range controllerProps {
+			if k == "DrbdOptions/auto-quorum" {
+				if v != "disabled" {
+					controllerQuorum = true
+				}
+			}
+		}
+		useQuorum := volQuorum || controllerQuorum
```
Comment (WanzenBug) [inline pkg/client/linstor.go:511]: We don't need to check the controller properties here. 

First of all we would also need to check the resource group property.

Secondly, this setting does not influence _existing_ resources, so only whatever is set directly on the resource definition is relevant.

So: just remove this and use `volQuorum` directly.

File: pkg/client/linstor.go [hunk 2]
```diff
@@ -227,6 +233,7 @@ func (s *Linstor) resourceDefinitionToVolume(resDef lapi.ResourceDefinitionWithV
 		ResourceGroup: resDef.ResourceGroupName,
 		FsType:        fsType,
 		Properties:    props,
+		UseQuorum:     quorum,
```
Comment (WanzenBug) [inline pkg/client/linstor.go:236]:
```suggestion
		UseQuorum:     resDef.Props["DrbdOptions/Resource/quorum"] != "off",
```

**Review by WanzenBug — CHANGES_REQUESTED** [review]
Please see that https://github.com/piraeusdatastore/linstor-csi/pull/163/checks pass.

Also some cleanup/simplification is needed. The logic change itself seems sound.

**Review by WanzenBug — APPROVED** [review]
LGTM. I'll run some e2e tests on this, but I don't expect any issues.

**Comment by rck** [comment]
@WanzenBug you squash it before merging, right?

**Comment by WanzenBug** [comment]
@tw185112 thank you for your contribution!

**Pull request merged**